src/main/java/com/adyen/httpclient/SSLContextFactory.java

```diff
@@ -0,0 +1,77 @@
+package com.adyen.httpclient;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+
+
+/**
+ * Utility class for creating {@link SSLContext} for different use cases.
+ */
+public final class SSLContextFactory {
+    private static final String TERMINAL_CERTIFICATE_ALIAS = "TerminalCertificate";
+    private static final String ANDROID_CA_STORE = "AndroidCAStore";
+    private static final String TLSV1_2 = "TLSv1.2";
+
+    public static SSLContext getClientCertificateAuthSSLContext(
+            KeyStore trustKeyStore,
+            String clientKeyStorePassword,
+            KeyStore clientKeyStore
+    ) throws GeneralSecurityException {
+        char[] password = null;
+        if (clientKeyStorePassword != null && !clientKeyStorePassword.isEmpty()) {
+            password = clientKeyStorePassword.toCharArray();
+        }
+
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory
```

Comment:
There is a lot of repetitive code in this class. Can we move this to a separate function that's called by all 3 public methods?